Userland/Libraries/LibJS/Runtime/ArrayPrototype.cpp

```diff
@@ -1034,4 +1035,42 @@ JS_DEFINE_NATIVE_FUNCTION(ArrayPrototype::values)
     return ArrayIterator::create(global_object, this_object, Object::PropertyKind::Value);
 }
 
+static void recursive_array_flat(GlobalObject& global_object, JS::Array& new_array, JS::Array& array, i32 depth, bool is_infinity)
```

Comment:
No need for `JS::` namespace prefix here, `array` should be an `Object` (e.g. `array_like`)